File: lab_python_oop/rectangle.py
from lab_python_oop.shape import GeometricShape
from lab_python_oop.colour import Colour

class Rectangle(GeometricShape):

    FIGURE_TYPE = "Прямоугольник"

    @classmethod
    def getFigureType(class_v):
        return class_v.FIGURE_TYPE

    def __init__(self, w, h, col):
        self._width = w
        self._height = h
        self._colour = Colour()
        self._colour.shapeColour = col
    
    @property
    def Width(self):
        return self._width

    @Width.setter
    def Width(self, w):
        self._width = w

    @Width.deleter
    def Radius(self, w):
        del self._width

    @property
    def Height(self):
        return self._height

    @Height.setter
    def Height(self, h):
        self._height = h

    @Height.deleter
    def Height(self, h):
        del self._height

    def Area(self):
        return self.Width * self.Height

    def __repr__(self) :
        return '{} cо сторонами {} и {}, площадь равна {}'.format(
            Rectangle.getFigureType(),
            self.Width,
            self.Height,
            self.Area()
            )

File: figures/rectangle.cs
namespace class_lab3
{
    /// <summary>
    /// Прямоугольник
    /// </summary>
    class Rectangle : GeometricShape, IPrint
    {
        /// <summary>
        /// Конструктор по двум параметрам
        /// </summary>
        /// <param name="h">Высота</param>
        /// <param name="w">Ширина</param>
        public Rectangle(double h, double w)
        {
            this.Height = h;
            this.Width = w;
        }
        /// <summary>
        /// Высота прямоугольника
        /// </summary>
        public double Height { get; set; }
        /// <summary>
        /// Ширина прямоугольника
        /// </summary>
        public double Width { get; set; }

        public override double Area()
        {
            return Height * Width;
        }

        public override string ToString()
        {
            return ("Прямоугольник со сторонами " + Height.ToString() + " и " + Width.ToString()
                + "; площадь: " + this.Area());
        }

        //public void Print()
        //{
        //    Console.WriteLine(this.ToString());
        //}
    }
}